.github/workflows/release.yml

```diff
@@ -1,96 +1,142 @@
+        uses: actions/checkout@v3
         with:
           fetch-depth: 0
-          ref: ${{ github.event.client_payload.branch }}
-      - uses: actions/setup-python@v1
-        with:
-          python-version: '3.7'
-      - name: Build and release docker images
+
+      - name: 📩 Checkout the Source
         run: |
-          rm -rf /usr/share/dotnet &
-          pip install -r requirements.txt
-          version=$(python -m armory --version)
-          docker login --username ${{ secrets.DOCKER_USERNAME }} --password ${{ secrets.DOCKER_PASSWORD }}
-          docker pull twosixarmory/pytorch:latest
-          python docker/build.py pytorch
-          docker push twosixarmory/pytorch:${version}
-          docker tag twosixarmory/pytorch:${version} twosixarmory/pytorch:latest
-          docker push twosixarmory/pytorch:latest
-  release-pytorch-deepspeech-docker:
-    name: Build and release pytorch-deepspeech docker image
-    needs: [release-wheel]
-    runs-on: ubuntu-18.04
-    steps:
-      - uses: actions/checkout@v2
-        with:
-          fetch-depth: 0
-          ref: ${{ github.event.client_payload.branch }}
-      - uses: actions/setup-python@v1
+          git fetch --prune --unshallow || true
+          git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+          # Verify .git folder
+          if [ ! -d "./.git" ]; then
+            echo "Expected ./.git folder to exist"
+            exit 1
+          fi
+
+      - name: 🌎 Setup Build Environment
+        run: |
+          pip install pip>=22.2.2
+          pip install .
+
+          armory configure --use-defaults
+
+          ARMORY_VERSION=`python -m armory --version 2> /dev/null | sed -r 's/\+/\./g'`
+          echo "ARMORY_VERSION=${ARMORY_VERSION}" >> $GITHUB_ENV
+
+      - name: ☁️ Login to DockerHub
+        if: ${{ env.RELEASE_DRY_RUN == 'false' }}
+        uses: docker/login-action@v1
         with:
-          python-version: '3.7'
-      - name: Build and release docker images
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+        # run: |
+        #   docker login -u ${{ secrets.DOCKER_USERNAME }} -p ${{ secrets.DOCKER_PASSWORD }}
+
+      # TODO: Build Base Image
+      - name: 🔨 Build Release Images
         run: |
-          rm -rf /usr/share/dotnet &
-          pip install -r requirements.txt
-          version=$(python -m armory --version)
-          docker login --username ${{ secrets.DOCKER_USERNAME }} --password ${{ secrets.DOCKER_PASSWORD }}
-          docker pull twosixarmory/pytorch-deepspeech:latest
-          python docker/build.py pytorch-deepspeech
-          docker push twosixarmory/pytorch-deepspeech:${version}
-          docker tag twosixarmory/pytorch-deepspeech:${version} twosixarmory/pytorch-deepspeech:latest
-          docker push twosixarmory/pytorch-deepspeech:latest
+          echo "Building Armory ${{ env.ARMORY_VERSION }} Images"
+          python docker/build.py --framework all
```

Comment:
Previously, this caused OOM errors on the runner.

Reply:
Updated to use matrix.